### PR TITLE
File parsing teams crud

### PR DESCRIPTION
**Pull request by taltaub22**
After making a mess from git,
I've fixed almost everything..
Waiting for your comments

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+let tournamentDataRouter = require('./routers/tournamentDataRouter');
+app.use('/tournamentData', tournamentDataRouter);
+
+let teamsRouter = require('./routers/teamsRouter');
+app.use('/team', teamsRouter)
+
+app.listen(appPort, () => {
+    //TODO: Add real correlation id
+    msLogger.info("corr-id","Server started on port " + appPort);
```
Comment (idanstark42) [inline index.js]: Change this to fit the automatic correlationId

File: logic/objectDataParser.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+function parseIntOrUndefined (int) {
+  if (int == '') {
+    return undefined
+  }
+  return parseInt(int)
+}
+
+function stringOrUndefined (string) {
+  if (string == '') {
```
Comment (idanstark42) [inline logic/objectDataParser.js]: The string `''` is falsly, so you can just do `string || undefined` and it would work

Comment (idanstark42) [reply logic/objectDataParser.js]: Same for all these functions

File: logic/objectDataParser.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const Match = require('../models/Match')
+const Team = require('../models/Team')
+const Table = require('../models/Table')
+const MatchTeam = require('../models/MatchTeam')
+
+function deserializeMatch (rawMatch) {
+  const newMatch = new Match()
+
+  newMatch.matchId = rawMatch[0]
```
Comment (idanstark42) [inline logic/objectDataParser.js]: Don't you have all of these properties in the constructor?

File: models/Team.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class Team {
+  constructor (number, name, affiliation, cityState, country, coach1, coach2, judgingGroup, pitNumber, pitLocation, translationNeeded) {
```
Comment (idanstark42) [inline models/Team.js]: These are alot of attribtues... I thin behond the *Must have* attributes, you can keep it ot external settings

File: routers/teamsRouter.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    connection.db().collection(TEAMS_COLLECTION).findOne({ 'number': parseInt(req.params.id) }).then(data => {
+      if (!data) {
+        res.sendStatus(404)
+        return
+      }
+
+      res.send(data)
+    })
+  }).catch(err => {
+    // TODO: use ms-logger
```
Comment (idanstark42) [inline routers/teamsRouter.js]: You already have ms-logger in your code

File: routers/teamsRouter.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    console.log(err)
+    res.sendStatus(500)
+  })
+})
+
+router.put('/:id', (req, res) => {
+  MongoClient.connect(MONGU_URI).then(connection => {
+    connection.db().collection(TEAMS_COLLECTION).findOneAndUpdate({ 'number': parseInt(req.params.id) }, req.body).then(a => {
+      if (a.ok === 1) {
```
Comment (idanstark42) [inline routers/teamsRouter.js]: a?

File: routers/tournamentDataRouter.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  MongoClient.connect(process.env.MONGO).then(conn => {
+    conn.db().collection('tables').insertMany(data.tables).then(() => {
+      // TODO: use ms-logger
+      console.log('all good')
+    }).catch(err => {
+      // TODO: use ms-logger
+      console.log(err)
+    })
+
+    conn.db().collection('teams').insertMany(data.teams).then(() => {
```
Comment (idanstark42) [inline routers/tournamentDataRouter.js]: Are you sure you want to insert the data from the user without validations first?

File: index.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+'use strict'
+const express = require('express');
+const msLogger = require('@first-lego-league/ms-logger')();
+msLogger.setLogLevel(msLogger.LOG_LEVELS.DEBUG);
```
[the remark this answers is not in the document]
Comment (taltaub22) [reply index.js]: Waiting for your comments in https://github.com/FirstLegoLeagueIL/architecture/issues/15

File: index.js (hunk 1 above)
Comment (taltaub22) [reply index.js]: Did you re-publish the package?

File: logic/objectDataParser.js (hunk 3 above)
Comment (taltaub22) [reply logic/objectDataParser.js]: Yes, I do.
In this way its more readable

File: routers/tournamentDataRouter.js (hunk 7 above)
Comment (taltaub22) [reply routers/tournamentDataRouter.js]: Validation is happening in the parser

File: routers/teamsRouter.js (hunk 5 above)
Comment (taltaub22) [reply routers/teamsRouter.js]: Waiting for you to publish the updated version with the correlation-id

File: index.js (hunk 1 above)
Comment (idanstark42) [reply index.js]: Waiting for your comments on the middleware PR

File: logic/objectDataParser.js (hunk 2 above)
Comment (idanstark42) [reply logic/objectDataParser.js]: I'm that can you can use `isNaN` or `isFinite`, but doing a simple `||` operation would also work here

File: logic/tournamentDataParser.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        Math.ceil(parseFloat(numOfPracticeMatches[1])))
+
+  const numOfTables = lines[blocks.find(x => x.blockId === RANKING_MATCH_SCHEDULE_ID).lineNumber + 2][1]
+  const numOfTeamsPerTable = lines[blocks.find(x => x.blockId === RANKING_MATCH_SCHEDULE_ID).lineNumber + 3][1]
+  const numOfActualFields = parseInt(numOfTables) * parseInt(numOfTeamsPerTable)
+  const tablesRaw = lines[blocks.find(x => x.blockId === RANKING_MATCH_SCHEDULE_ID).lineNumber + TABLE_NAMES_LINE]
+
+  const tables = []
+  for (let i = 3; i < numOfActualFields + 3; i++) {
+    tables.push(new Table(i - 3, tablesRaw[i]))
```
Comment (idanstark42) [inline logic/tournamentDataParser.js]: This is much less readable then givving this 3 a name to represent it like the other constants, and just use `tableRaw[i+varName]`

**Pull request merged**